modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/utils/getFields.ts

```diff
@@ -23,127 +28,110 @@ interface IProperty {
 	extensions?: any;
 	format?: EFieldFormat;
 	items?: any;
-	type: EFieldType;
+	type?: EFieldType;
 }
 
 interface IProperties {
 	[key: string]: IProperty;
 }
 
-interface ISchemas {
+export interface ISchemas {
 	[key: string]: {
 		properties: IProperties;
 		type: string;
 	};
 }
 
-function getValidFields({
+const filterSchemaProperty = (propertyKey: string) => {
+	return (
+		!INVALID_FIELDS.includes(propertyKey) &&
+		!propertyKey.includes(LOCALIZABLE_PROPERTY_SUFFIX)
+	);
+};
+
+export function getValidFields({
 	contextPath,
 	schemaName,
+	schemaStack,
 	schemas,
 }: {
 	contextPath: string;
 	schemaName: string;
+	schemaStack: string[];
 	schemas: ISchemas;
 }): Array<IField> {
 	const fields: Array<IField> = [];
 
 	const properties: IProperties = schemas[schemaName]?.properties;
 
-	properties &&
-		Object.keys(properties).map((propertyKey) => {
-			const propertyValue = properties[propertyKey];
-
-			if (INVALID_FIELDS.includes(propertyKey)) {
-				return;
-			}
+	if (!properties) {
+		return fields;
+	}
 
-			if (propertyKey.includes(LOCALIZABLE_PROPERTY_SUFFIX)) {
-				return;
-			}
+	Object.keys(properties)
+		.filter(filterSchemaProperty)
```

Comment:
We can keep this logic inline, to keep code close and avoid naming awkwarness.

If you prefer keeping it separate, I'd call this something like `validSchemaPropertyFilter`, and rename `INVALID_FIELDS` to something like `BLACKLISTED_FIELDS`, to avoid `valid` in two places.